src/backend/preprocessor/Preprocessor.js

```diff
@@ -0,0 +1,89 @@
+define([
+    './cfgGenerator',
+    './envGenerator'
+], function(cfgGenerator, envGenerator) {
+
+    var _ = require('underscore');
```

Comment:
this will only work in node.js environment. Please require in a RequireJS style (add a bower dep if necessary). Furthermore, if you need a underscore-like library, please use lodash because it's newer and seemingly faster.